GVFS/GVFS.Service/RepoRegistry.cs

```diff
@@ -166,27 +175,24 @@ public void AutoMountRepos(int sessionId)
         {
             using (ITracer activity = this.tracer.StartActivity("AutoMount", EventLevel.Informational))
             {
-                using (GVFSMountProcess process = new GVFSMountProcess(activity, sessionId))
+                List<RepoRegistration> activeRepos = this.GetActiveReposForUser(this.gvfsPlatform.GetUserIdFromLoginSessionId(sessionId, activity));

```

Comment:
I think if we have this `AutoMountRepos` take the user id instead of the `sessionId` we could remove the GVFSPlatform as a dependency of this class.  We should be able to use the `this.fileSystem` instead of the `this.gvfsPlatform.FileSystem` below right?  Or is that a different file system?  If so I think we have issues having 2 file systems and this class has and uses both.  That is and will be a point of confusion for sure.

Reply:
>>We should be able to use the this.fileSystem instead of the this.gvfsPlatform.FileSystem below right? Or is that a different file system

We still need `this.gvfsPlatform.FileSystem` for `TryGetNormalizedPath()`. `TryGetNormalizedPath` is not available in `this.fileSystem`.

Reply:
As discussed - added `TryGetNormalizedPath` method to `PhysicalFileSystem`. Also updated the callers of `AutoMountRepos` to generate `userId` from `sessionId`. This helped remove the dependency of `RepoRegistry` on `GVFSPlatform`. Now `RepoRegistry` only uses one `PhysicalFileSystem`.